pkg/controller/dns/cleanup_runner.go

```diff
@@ -0,0 +1,29 @@
+package dns
+
+import (
+	"context"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type CleanupRunner struct {
+	client        client.Client
+	namesToDelete []string
```

Comment:
unusedNameLabels would be a better name for this field

Reply:
done